### PR TITLE
feat: add `impl_from` argument to `#[server]` proc_macro

### PR DESCRIPTION
**Pull request by videobitva**
This PR adds an optional `impl_from` named  argument for `server_fn` proc macro.
This argument defaults to `true` and enables (or disables) the impl of `From` trait for server function argument type.

In cases when the server function has more than one argument, the `impl_from` has no effect on the trait impl generation, as it has been already disabled by function arguments number check.

The related issue has been discussed here: https://discord.com/channels/1031524867910148188/1203001939089035295

TL;DR: orphan rules disable (in some cases) the implementation of `From` trait for conversion between the only server function argument and server function argument type. 

The minimal example of the issue lives here: https://github.com/videobitva/test_server_fn

### CODE REVIEW

File: server_fn_macro/src/lib.rs [hunk 1]
```diff
@@ -119,6 +119,7 @@ pub fn server_macro_impl(
         res_ty,
         client,
         custom_wrapper,
+        impl_from: impl_into,
```
Comment (gbj) [inline server_fn_macro/src/lib.rs:122]: This line is what made me think it should be renamed: if what it does is `impl From<_>`, and we're using that language internally, it might make sense to do the same thing in the API.

Comment (videobitva) [reply server_fn_macro/src/lib.rs]: The funny thing is that initially it was named actually as `impl_from`, but later on I renamed it to be `impl_into` -- this line of code seems to be an artifact of that process. I've changed it back in the receing commit.

File: leptos_macro/src/lib.rs [hunk 2]
```diff
@@ -883,6 +883,9 @@ pub fn slot(args: proc_macro::TokenStream, s: TokenStream) -> TokenStream {
 ///     - `"GetCbor"`: `GET` request with URL-encoded arguments and CBOR response
 /// - `req` and `res` specify the HTTP request and response types to be used on the server (these
 ///   should usually only be necessary if you are integrating with a server other than Actix/Axum)
+/// - `impl_into`: specifies whether to implement trait Into for server function's argument or not,
```
[the remark this answers is not in the document]
Comment (videobitva) [reply leptos_macro/src/lib.rs]: I've renamed the argument back and merged the initial description with your variant, hope it is better now.

**Comment by gbj** [comment]
Excellent, thanks!
